shapely/speedups/_speedups.pyx

```diff
@@ -49,11 +50,11 @@ def geos_linestring_from_py(ob, update_geom=None, update_ndim=0):
             n = 2
 
         if type(ob) == LineString:
-            return <unsigned long>GEOSGeom_clone_r(handle, g), n
+            return <uintptr_t>GEOSGeom_clone_r(handle, g), n
         else:
-            cs = GEOSGeom_getCoordSeq_r(handle, g)
+            cs = <GEOSCoordSequence*>GEOSGeom_getCoordSeq_r(handle, g)
```

Comment:
@pelson The warnings look like this:

```
shapely/speedups/_speedups.c:1694:18: warning: assigning to
      'GEOSCoordSequence *' (aka 'struct GEOSCoordSeq_t *') from 'const
      GEOSCoordSequence *' (aka 'const struct GEOSCoordSeq_t *') discards
      qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
      __pyx_v_cs = GEOSGeom_getCoordSeq_r(__pyx_v_handle, __pyx_v_g);
                 ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

It's because the coordinate sequence returned by `GEOSGeom_getCoordSeq_r` is const (geometries should be immutable), while the coordinate sequence returned by `GEOSCoordSeq_create_r` is mutable because it hasn't finished being defined.

https://github.com/libgeos/libgeos/blob/e95fcc7d8b9b6b58ae1a04ff889bcf62c0c0b15b/capi/geos_ts_c.cpp#L3776-L3777

https://github.com/libgeos/libgeos/blob/e95fcc7d8b9b6b58ae1a04ff889bcf62c0c0b15b/capi/geos_ts_c.cpp#L3488-3489